peachjam/models/user_following.py

```diff
@@ -290,8 +306,48 @@ def _update_search(self):
         self.save(update_fields=["last_alerted_at"])
         return True
 
+    def _update_new_citation(self, citation):
+        if not self.saved_document:
+            log.error("User %s follow %s is not for a saved document", self.user, self)
+            return
+
+        # check that we are passing a citation to the saved document
+        if citation.target_work != self.saved_document.work:
+            log.error(
+                "Citation target work %s does not match saved document work %s",
+                citation.target_work,
+                self.saved_document.work,
+            )
+            return
+
+        # check if the user has ever been alerted about this citation
+        events = TimelineEvent.objects.filter(
```

Comment:
I suspect this could be boiled down to a single `TimelineEvent.objects.filter(...).exists()` which would be more efficient.